language/tools/move-package/src/source_package/parsed_manifest.rs

```diff
@@ -38,7 +39,21 @@ pub struct Dependency {
     pub local: PathBuf,
     pub subst: Option<Substitution>,
     pub version: Option<Version>,
-    pub digest: Option<Vec<u8>>,
+    pub digest: Option<PackageDigest>,
+    pub git_info: Option<GitInfo>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct GitInfo {
+    /// The git clone url to download from
+    pub git_url: Symbol,
+    /// The git reference -- a commit SHA or a branch name, e.g., 'main'
+    pub git_ref: Symbol,
+    /// The path under this repo where the move package can be found -- e.g.,
+    /// 'language/move-stdlib`
+    pub subdir: PathBuf,
+    /// Where the git repo is downloaded to.
+    pub download_to: PathBuf,
```

Comment:
Is this controlled by the package manager?

Reply:
Yes. The `download_to` is controlled by the package manager and is based off of the package name and specific reference for the git repo and will be of the form `<pkg_name>_<git_ref/branch>`, e.g., `MoveStdlib_main` or `MoveStdlib_ 96d28e132dac33542bdafa4ac25324015d083bf1`

Reply:
Would it make more sense to base it off of the repo url provided? That way you could reuse the same repo for N packages (if N move packages happen to be in that repo) instead of needing to clone the whole repo N times?

Reply:
I like it :)  I've updated it to be based off the repo url so we can reuse the repo for multiple packages.

Reply:
One caveat with this change is that you may want to have some logic at resolve time to force that all references into a repo are done at one revision, or have some capabilities for multiple checkouts at different revisions for 1 repository (this may be more challenging than the former)

Reply:
We currently tag the checkout by the revision/branch name that is being checked out. So you can absolutely have one package on your system using `MoveStdlib` tracking `main` and another fixed at `ef45123...` etc.